### ISSUE TITLE
mirage.io updates/improvements

### ISSUE THREAD

**Issue by GemmaG** [mirage/mirage#569]
In general: 
- [ ] generate more interesting/non trivial use cases for mirage/unikernels beyond just setting up a static site etc. Perhaps use an existing OCaml project (e.g. Eliom) and use with with a unikernel setup? Also multiple unikernel operations for projects...

Homepage: https://mirage.io/
- [ ] Add information about the next Mirage meeting/irc on the homepage, top right
- [ ] recent updates - include fewer/link to more blog posts (currently a long list of IRC notes)
- [ ] link to canopy from the homepage - initiate wider use of it

Pioneer Projects: https://github.com/mirage/mirage-www/wiki/Pioneer-Projects
- [ ] Link from homepage? So people can see how to get involved immediately
- [ ] Are their difficulty ratings accurate? Shall we create some easier ones to get people started? Perhaps having them in difficulty order could work.
- [ ] keep them up to date more often, who is working on them.

Docs: https://mirage.io/docs/
- [ ] "libraries" section is not super useful as a lot of it is out of date... lwt, portable regular expression, COW, ATcaml etc is misleading and not helpful
- [ ] create a survey of library Readmes so that someone can check down the list to see if they are relevant/up to date/not confusing/detrimental (has been listed as a possible project, but people might not take it up
- [ ] move the "getting started" menu to the right, where the other menu is sitting at the moment
- [ ] Tutorials: update current tutorials, add new screencasts where appropriate (short and relevant). Combination of text and cast/video is good.
- [ ] improve the cross linking between library readmes on github and the mirage docs area (on website) to make the process of finding info easier/more streamlined

Community: https://mirage.io/community/
- [ ] remove out-of-date information:
- [ ] community: face to face sessions at makespace, update twitter paragraph
- [ ] make the code of conduct more visible. Currently "Contributing to MirageOS"

Releases: https://mirage.io/releases/
- [ ] changelog: still useful? Not been updated in a while
- [ ] link blog: needs updating

**Comment by Drup** [comment]
I'm just going to answer the "eliom" part (the rest is great): No for now, maybe in a few month. Even if it's possible (currently, it's not), the build system aspect is going to be .. interesting.

**Comment by yomimono** [comment]
Can we rename "pioneer projects"? While the alliteration is attractive, the meaning is a bit muddy.  I think the intent is that these are projects with well-defined boundaries, which someone could contribute to without getting overwhelmed in dependencies or wider systems; maybe we can come up with a better way to convey that.

**Comment by yomimono** [comment]
What we have at "contributing to MirageOS" isn't a code of conduct by any means, but rather a set of instructions for reporting issues, submitting changes, and reaching out to the MirageOS community.

**Comment by yomimono** [comment]
Also, thanks very much @GemmaG for submitting this issue -- these are important issues for us to address, and I appreciate your effort in compiling them and all the work you've done to address similar issues so far.

**Comment by avsm** [comment]
I'll second @yomimono -- these are important and gating issues for MirageOS 3.  Some specific responses:

* On the changelog: I have been building [twiopam](https://github.com/avsm/twiopam) to collate the OPAM movements and publish them regularly.  This will be the successor to the current very rough changelog.  Examples of early output are up at https://gist.github.com/bactrian, but are still being tweaked.
* Link blog: I don't think maintaining this inline on the website is sustainable. How about moving the link blog over to a collaborative service such as [Pinboard](http://pinboard.in) and mirroring it to the website via API sync? @amirmc?
* Pioneer projects: I agree that we need to surface and link these more explicitly from the main website.  On a related note, we do not currently directly link to Canopy.

**Comment by talex5** [comment]
I agree with most of this, but I'd like to point out that we updated the Lwt introduction quite recently, so hopefully it isn't out-of-date. Please let me know if you found problems with it.

**Comment by avsm** [comment]
On the topic of simple mirage projects, we could:

- describe how the self-hosted infrastructure works.  This is mirage.io, decks.openmirage.org (which needs to be migrated to mirage.io), and hopefully soon a solo5.mirage.io and DNS server.  The build infrastructure uses Docker Cloud to abstract builds.
- [mirage-seal](https://github.com/mirage/mirage-seal) is a way to rapidly build a static website, but needs to be updated.
- There are several ongoing intern projects by @matildah and @wiredsister about NTP and Syslogd daemons, along with their blog posts and documentation.

**Comment by hannesm** [comment]
related is https://github.com/mirage/mirage-www/issues/432 ... I believe we can fix minor things all the time, but maybe should first have a bigger picture in mind?
related for basic unikernels might be https://github.com/cfcs/mirage-examples

**Comment by amirmc** [comment]
Some fairly quick and rough thoughts:

- Great list, thanks for compiling it! :)
- I agree that these are important but I'm wary of saying that they're all 'gating' (i.e. should *block release* of 3.0). We should identify which are 'must have' and which are 'nice to have'.
- **Use cases**: Non-trivial use-cases would be great.  We do have non-static site pieces in skeleton and could highlight those. There's also @talex5's work on his [REST service](http://roscidus.com/blog/blog/2014/07/28/my-first-unikernel/) and also the [QubesOS unikernel firewall](http://roscidus.com/blog/blog/2016/01/01/a-unikernel-firewall-for-qubesos/). These could be highlighted on the site more prominently.
- **Pioneer Projects**: I'd rather we didn't rename the pioneer projects unless there's a very compelling reason. For example, I don't really consider "GSoC" or "Outreachy" to be particularly clear either.  However, we _should_ make it clearer what they are so I've added a note to the top of the [projects page](https://github.com/mirage/mirage-www/wiki/Pioneer-Projects) based on @yomimono's comment.
- **Link blog**: I'm not sure how valuable this is anymore.  It was helpful for us in the early days but it's tedious to maintain.  We can set up something via Pinboard but will still require some tooling to consume an RSS feed and present it in a page (this code likely already exists somewhere). I personally wouldn't consider this item a priority though (maybe embedding a twitter favourites feed is better?).
- **Surfacing projects**: I think we should move projects into issue trackers and put together some tooling to aggregate them on the website. This would be useful in a more general sense (e.g. see [unikernel.org#30](https://github.com/Unikernel-Systems/unikernel.org/issues/30)).

**Comment by hannesm** [comment]
about the pioneer projects: I find that wiki page way too huge and would suggest to move them to canopy (into a subdirectory `wanted`, one project per page).  we can use tags (or prefixed tags `wanted:network` `wanted:protocol` etc.) to get an overview and cross links.

**Comment by yomimono** [comment]
A lot of these are still good ideas!  Can we get an update from @avsm and @GemmaG  about which of them are addressed in the forthcoming `bushel` and which could still use some attention?

**Comment by hannesm** [comment]
I wonder: this is an issue that is several years old. The website recently got some updates. Is it worth to keep this issue? Should smaller issues be opened for the mirage-www repository?
